pkg/build/api/validation/validation_test.go

```diff
@@ -1008,10 +1008,11 @@ func TestValidateTrigger(t *testing.T) {
 			trigger:  buildapi.BuildTriggerPolicy{},
 			expected: []*fielderrors.ValidationError{fielderrors.NewFieldRequired("type")},
 		},
-		"trigger with unknown type is valid, but ignored": {
+		"trigger with unknown type": {
 			trigger: buildapi.BuildTriggerPolicy{
 				Type: "UnknownTriggerType",
 			},
+			expected: []*fielderrors.ValidationError{fielderrors.NewFieldInvalid("type", "", "")},
```

Comment:
See the test below which exercises the full stack and ensures a build config with an unknown type can be submitted